data-runtime/src/main/java/io/micronaut/data/runtime/intercept/criteria/AbstractSpecificationInterceptor.java

```diff
@@ -156,6 +158,15 @@ protected final <E, QR> PreparedQuery<E, QR> preparedQueryForCriteria(Repository
         return preparedQueryDecorator.decorate(preparedQuery);
     }
 
+    private <E> StoredQuery<E, ?> buildExists(MethodInvocationContext<T, R> context, QueryBuilder sqlQueryBuilder) {
+        Class<E> rootEntity = getRequiredRootEntity(context);
+        CriteriaUpdateBuilder<E> criteriaUpdateBuilder = getCriteriaUpdateBuilder(context);
```

Comment:
`CriteriaUpdateBuilder` doesn't looks correct

Reply:
Yeah, I noticed this too. I'm currently trying to implement it right.